### PR TITLE
chore(python): update noxfile.py and .kokoro/requirements.txt

### PR DESCRIPTION
**Pull request by parthea**
This PR contains changes to both `.kokoro/requirements.txt` and `noxfile.py`. The changes are combined into a single PR in order to reduce the number of PRs downstream as each commit to synthtool triggers a PR to be created in downstream repositories in the `googleapis` organization.

The changes to `.kokoro/requirements.txt` were created by deleting the [requirements.txt file](https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/.kokoro/requirements.txt) and running the following command `pip-compile --allow-unsafe --generate-hashes requirements.in`

The changes to `noxfile.py` were tested by following tests steps

1. Build the post processor image

For the split repo, run `docker build -f docker/owlbot/python/Dockerfile -t test_split_repo .`
For the mono repo, run `docker build -f docker/owlbot/python_mono_repo/Dockerfile -t test_mono_repo .`

2. Run the post processor image. In the downstream repo, run the post processor image and nox sessions.

For example, in python-pubsub, run `docker run --user $(id -u):$(id -g) --rm -v $(pwd):/repo -w /repo test_split_repo`, then run nox sessions `nox -s unit-3.9` and `nox -s prerelease_deps`

For example, in the monorepo google-cloud-python run ` mkdir -p owl-bot-staging/google-cloud-access-approval` and `docker run --user $(id -u):$(id -g) --rm -v $(pwd):/repo -w /repo test_mono_repo` then run nox sessions `nox -s unit-3.9` and `nox -s prerelease_deps` from the directory `packages/google-cloud-access-approval`

3. Confirm that nox sessions pass and all protobuf implementations are tested including `upb`, `python` and `cpp`

### CODE REVIEW

**Review by ohmayr — APPROVED** [review]

File: synthtool/gcp/templates/python_library/noxfile.py.j2 [hunk 1]
```diff
@@ -227,14 +227,27 @@ def install_unittest_dependencies(session, *constraints):
...
 
+    if protobuf_implementation == "cpp" and session.python in ("3.11", "3.12"):
+        session.skip("cpp implementation is not supported in python 3.11+")
+
     constraints_path = str(
         CURRENT_DIRECTORY / "testing" / f"constraints-{session.python}.txt"
     )
     install_unittest_dependencies(session, "-c", constraints_path)
 
+    # Remove the 'cpp' implementation once support for Protobuf 3.x is dropped.
```
Comment (vchudnov-g) [inline synthtool/gcp/templates/python_library/noxfile.py.j2:246]: nit: Add a `TODO` prefix to this comment.

File: synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2 [hunk 2]
```diff
@@ -232,14 +232,27 @@ def install_unittest_dependencies(session, *constraints):
...
 
+    if protobuf_implementation == "cpp" and session.python in ("3.11", "3.12"):
+        session.skip("cpp implementation is not supported in python 3.11+")
+
     constraints_path = str(
         CURRENT_DIRECTORY / "testing" / f"constraints-{session.python}.txt"
     )
     install_unittest_dependencies(session, "-c", constraints_path)
 
+    # Remove the 'cpp' implementation once support for Protobuf 3.x is dropped.
```
Comment (vchudnov-g) [inline synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2:251]: nit: add `TODO`

**Review by vchudnov-g — COMMENTED** [review]
lgtm. Just a couple of nits.

File: synthtool/gcp/templates/python_library/noxfile.py.j2 (hunk 1 above)
Comment (parthea) [reply synthtool/gcp/templates/python_library/noxfile.py.j2:246]: Done in https://github.com/googleapis/synthtool/pull/1977

File: synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2 (hunk 2 above)
Comment (parthea) [reply synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2:251]: Done in https://github.com/googleapis/synthtool/pull/1977

**Pull request merged**